*** target/generated-sources/com/github/albrechtf/pmd/lang/vb/ast/ASTProcDeclaration.java	2016-09-20 17:10:39.958678200 +0200
--- src/gen/java-customized-ast/ASTProcDeclaration.java	2015-09-08 16:25:20.817205000 +0200
***************
*** 11,16 ****
--- 11,71 ----
    public ASTProcDeclaration(VisualBasicParser p, int id) {
      super(p, id);
    }
+   
+ 	/**
+ 	**	The type of the procedure: Sub, Function, Property
+ 	*/
+ 	private int procType;
+ 	public void setProcType(int procType)
+ 	{
+ 		this.procType = procType;
+ 	}
+ 	public int getProcType()
+ 	{
+ 		return procType;
+ 	}
+ 
+ 	// The name is the first child of type Name.
+ 	public String getName()
+ 	{
+ 		for (int i = 0; i < jjtGetNumChildren(); i++)
+ 		{
+ 			if (jjtGetChild(i) instanceof ASTName)
+ 				return jjtGetChild(i).getName();
+ 		}
+ 		
+ 		return null;
+ 	}
+ 
+ 	// The formal parameter list (if any) will be second
+ 	public ASTFormalParamList getParams()
+ 	{
+ 		for (int i = 0; i < jjtGetNumChildren(); i++)
+ 		{
+ 			if (jjtGetChild(i) instanceof ASTFormalParamList)
+ 				return (ASTFormalParamList)jjtGetChild(i);
+ 		}
+ 
+ 		return null;
+ 	}
+ 
+ 	// The function return type (if any) is next.
+ 	public ASTTypeName getReturnType()
+ 	{
+ 		for (int i = 0; i < jjtGetNumChildren(); i++)
+ 		{
+ 			if (jjtGetChild(i) instanceof ASTTypeName)
+ 				return (ASTTypeName)jjtGetChild(i);
+ 		}
+ 
+ 		return null;
+ 	}
+ 
+ 	// The Statements are always the last entry
+ 	public ASTStatements getStatements()
+ 	{
+ 		return (ASTStatements)jjtGetChild(jjtGetNumChildren()-1);
+ 	}  
  
  
    /** Accept the visitor. **/
